### PR TITLE
subscriptions: test cleanup

### PR DESCRIPTION
**Pull request by prateekbh**
- cleanup tests for subscriptions

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/entitlement.js [hunk 1]
```diff
@@ -101,6 +102,9 @@ export class Entitlement {
    * @return {boolean}
    */
   enablesThis() {
+    if (!this.product_) {
+      dev().warn('amp-subscriptions', 'Current Product is not set');
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement.js]: SMH, this feels wrong. This should be a dev assert - i.e. if this happens, this is just our bug.

File: extensions/amp-subscriptions/0.1/test/test-amp-subscriptions.js [hunk 2]
```diff
@@ -126,8 +129,21 @@ describes.realWin('amp-subscriptions', {amp: true}, env => {
...
+    const platform = new SubscriptionPlatform();
+    const entitlementData = {source: 'local',
+      service: 'local', products, subscriptionToken: 'token'};
+    const entitlement = Entitlement.parseFromJson(entitlementData);
+    const factoryStub = sandbox.stub().callsFake(() => platform);
+
+    subscriptionService.platformStore_ = new PlatformStore(
+        [serviceData.serviceId]);
+
+    platform.getEntitlements = () => Promise.resolve(entitlement);
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/test/test-amp-subscriptions.js]: Use `sandbox.stub()`

File: extensions/amp-subscriptions/0.1/test/test-platform-store.js [hunk 3]
```diff
@@ -257,16 +256,11 @@ describes.realWin('Platform store', {}, () => {
...
-    });
 
-    // TODO(prateekbh, #14336): Fails due to console errors.
-    it.skip('should report fatal error if all platforms fail', () => {
+    it('should report fatal error if all platforms fail', () => {
       platformStore.reportPlatformFailure('service1');
-      platformStore.reportPlatformFailure('service2');
-      expect(errorSpy).to.be.calledOnce;
+      expect(() => platformStore.reportPlatformFailure('service2'))
+          .to.throw(/All platforms have failed to resolve/);
```
Comment (rsimha) [inline extensions/amp-subscriptions/0.1/test/test-platform-store.js]: @prateekbh FYI, this test can now be fixed more easily using `allowConsoleError`. I've reverted this change and applied the simple fix in an upcoming change.

**Comment by prateekbh** [comment]
@rsimha I thought `allowConsoleError` was a temporary fix until everything is not fixed, and that adding `expect(...).to.throw(...)` to the test was the right thing to do.

Am i wrong?

**Comment by rsimha** [comment]
@prateekbh `allowConsoleError` is a pattern that will permanently be available to allow tests to specifically induce and verify errors. In no other cases should we see errors during tests.

The thing that's temporary is that for now, tests will print a message with the error, but continue to pass. Once all tests are cleaned up, we will go back to failing tests with unexpected errors.

Hope that makes sense.

**Pull request merged**